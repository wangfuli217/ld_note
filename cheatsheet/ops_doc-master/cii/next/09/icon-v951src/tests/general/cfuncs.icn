#  Test dynamic loading with a few standard cfuncs.

link cfunc

procedure main()
   local i

   every i := 500 to 513 do
      gen(bitcount, i)

   write()
   gen(vword, 314159)
   gen(lgconv, 10 ^ 30)
   gen(unpack, "abcd")
   gen(pack, 1684234849)
end

procedure gen(p, a)
   write(image(p), "(", image(a), ") = ", p(a) | "[failed]")
   return
end
