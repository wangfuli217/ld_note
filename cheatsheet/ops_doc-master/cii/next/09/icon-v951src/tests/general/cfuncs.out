procedure bitcount(500) = 6
function bitcount(501) = 7
function bitcount(502) = 7
function bitcount(503) = 8
function bitcount(504) = 6
function bitcount(505) = 7
function bitcount(506) = 7
function bitcount(507) = 8
function bitcount(508) = 7
function bitcount(509) = 8
function bitcount(510) = 8
function bitcount(511) = 9
function bitcount(512) = 1
function bitcount(513) = 2

procedure vword(314159) = 314159
procedure lgconv(integer(~10^30)) = 1000000000000000000000000000000
procedure unpack("abcd") = 1684234849
procedure pack(1684234849) = abcd
